lib/models/post.js

```diff
@@ -54,7 +55,7 @@ module.exports = ctx => {
     const { config } = ctx;
     let partial_url = self.url_for(this.path);
     if (config.relative_link) partial_url = `/${partial_url}`;
-    return config.url + partial_url.replace(config.root, '/');
+    return encodeURL(config.url + partial_url.replace(config.root, '/'));
```

Comment:
This should use `full_url_for()` as well.

Reply:
`url_for` is necessary for `config.relative_link`. Frankly, I'm not exactly sure why it's there in the first place (why it prepends a slash), so I retain it.

Reply:
@curbengh
`config.url + '/' + url_for(this.path).replace(config.root, '/')` is what developer used to get absolute path.
With full_url_for() helper it is not needed any more.

```js
  Post.virtual('permalink').get(function() {
    let url = full_url_for.call(this.path);
    if (config.pretty_urls.trailing_index === false) url = url.replace(/index\.html$/, '');
    return url
```

Reply:
```yaml
url: https://example.com/blog
root: /blog/
relative_link: true
```

```js
let partial_url = self.url_for('/path/to/a');
// partial_url = '/blog/path/to/a'
if (config.relative_link) partial_url = `/${partial_url}`;
// partial_url = '//blog/path/to/a'
return config.url + partial_url.replace(config.root, '/');
// https://example.com//blog/path/to/a
```

Reply:
my apology, I'm a bit slow, I just realized that

``` js
if (config.relative_link) partial_url = `/${partial_url}`;
```

is actually a workaround for `partial_url.replace(config.root, '/')`.

Ignore the following example, it's for myself

``` yml
url: https://example.com/blog
root: /blog/
```

``` js
// assuming this.path = /foo/bar
let partial_url = self.url_for(this.path);
// partial_url becomes '/blog/foo/bar'

if (config.relative_link) partial_url = `/${partial_url}`;
// partial url becomes '//blog/foo/bar'

return config.url + partial_url.replace(config.root, '/');
// 'https://example.com/blog' + '//blog/foo/bar'.replace('/blog/', '/')
// 'https://example.com/blog' + '/foo/bar'
// 'https://example.com/blog/foo/bar'
```

Without the workaround, it would results in 'https://example.com/blogfoo/bar'.

Reply:
Anyway, I've refactored.